### PR TITLE
[cpptrace] Add new port

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [ ] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/cpptrace/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "dependencies": [
+        {
+          "name": "vcpkg-cmake",
+          "host": true
+        },
+        {
+          "name": "vcpkg-cmake-config",
+          "host": true
+        }
+    ]
```
Comment (LilyWangLL) [inline ports/cpptrace/vcpkg.json:16]: Please use command `.\vcpkg.exe format-manifest ports/cpptrace/vcpkg.json` to format the file `vcpkg.json`.

File: ports/cpptrace/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,18 @@
+vcpkg_from_github(
+	OUT_SOURCE_PATH SOURCE_PATH
+	REPO jeremy-rifkin/cpptrace
+	REF 4b11b87e4d905d003d0325a53994441cc767017a
+	SHA512
+7800f38e765bf667abf6b91dff88041ec9f3d6bc94a98f0745c90cfa575b6694adbe9f620de9518ebab1b1cb65d2c3c16fc58d8fae23084a94d3806cd7901c5b
+	HEAD_REF main
```
Comment (LilyWangLL) [inline ports/cpptrace/portfile.cmake:7]:
```suggestion
    OUT_SOURCE_PATH SOURCE_PATH
    REPO jeremy-rifkin/cpptrace
    REF 4b11b87e4d905d003d0325a53994441cc767017a
    SHA512 7800f38e765bf667abf6b91dff88041ec9f3d6bc94a98f0745c90cfa575b6694adbe9f620de9518ebab1b1cb65d2c3c16fc58d8fae23084a94d3806cd7901c5b
    HEAD_REF main
```

File: ports/cpptrace/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+vcpkg_cmake_configure(
+	SOURCE_PATH "${SOURCE_PATH}"
```
Comment (LilyWangLL) [inline ports/cpptrace/portfile.cmake:11]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/cpptrace/portfile.cmake (hunk 2 above)
Comment (LilyWangLL) [reply ports/cpptrace/portfile.cmake:7]: Just some suggestions on formatting.

File: ports/cpptrace/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,18 @@
+vcpkg_from_github(
+	OUT_SOURCE_PATH SOURCE_PATH
+	REPO jeremy-rifkin/cpptrace
+	REF 4b11b87e4d905d003d0325a53994441cc767017a
```
Comment (dg0yt) [inline ports/cpptrace/portfile.cmake:4]:
```suggestion
	REF "v${VERSION}"
```
(Note that this requires a change to `SHA512`.)

File: ports/cpptrace/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+	"name": "cpptrace",
+	"version-string": "v0.1",
```
Comment (LilyWangLL) [inline ports/cpptrace/vcpkg.json:3]:
```suggestion
  "version": "0.1",
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Pinging @jeremy-rifkin for response. Is work still being done for this PR?

**Comment by jeremy-rifkin** [comment]
Hi @LilyWangLL, thank you so much for the comments! I've gotten sidetracked - I will close this and create a new PR when I'm able to return to this.

**Pull request closed without merging**